src/main/java/com/ss/mqtt/broker/model/topic/AbstractTopic.java

```diff
@@ -19,29 +19,28 @@
     static final String MULTI_LEVEL_WILDCARD = "#";
     static final String SINGLE_LEVEL_WILDCARD = "+";
 
-    static void checkTopic(@NotNull String topic) {
-        if (topic.length() == 0) {
-            throw new IllegalArgumentException("Topic has zero length.");
-        } else if (topic.contains("//") || topic.startsWith("/") || topic.endsWith("/")) {
-            throw new IllegalArgumentException("Topic has zero length level: " + topic);
-        }
+    static boolean checkTopic(@NotNull String topic) {
+        return topic.length() != 0 && !topic.contains("//") && !topic.startsWith("/") && !topic.endsWith("/");
     }
 
     private final @NotNull String[] segments;
     private final @NotNull String rawTopic;
+    private final @Getter boolean invalid;
```

Comment:
why do we need to have an additional flag in this class to understand that this topic is not valid?